packages/@webex/plugin-cc/src/services/core/WebSocket/connection-service.ts

```diff
@@ -0,0 +1,144 @@
+import {WebSocketManager} from './WebSocketManager';
+import {SubscribeRequest} from '../../../types';
+import LoggerProxy from '../../../logger-proxy';
+import {
+  LOST_CONNECTION_RECOVERY_TIMEOUT,
+  WS_DISCONNECT_ALLOWED,
+  CONNECTIVITY_CHECK_INTERVAL,
+} from '../constants';
+
+type ConnectionLostDetails = {
+  isConnectionLost: boolean;
+  isRestoreFailed: boolean;
+  isSocketReconnected: boolean;
+  isKeepAlive: boolean;
+};
+
+type ConnectionProp = {
+  lostConnectionRecoveryTimeout: number;
+};
+
+export class ConnectionService extends EventTarget {
+  private connectionProp: ConnectionProp = {
+    lostConnectionRecoveryTimeout: LOST_CONNECTION_RECOVERY_TIMEOUT,
+  };
+
+  private wsDisconnectAllowed = WS_DISCONNECT_ALLOWED;
+  private reconnectingTimer: ReturnType<typeof setTimeout>;
+  private restoreTimer: ReturnType<typeof setTimeout>;
+  private isConnectionLost: boolean;
+  private isRestoreFailed: boolean;
+  private isSocketReconnected: boolean;
+  private isKeepAlive: boolean;
+  private reconnectInterval: ReturnType<typeof setInterval>;
+  private webSocketManager: WebSocketManager;
+  private subscribeRequest: SubscribeRequest;
+
+  constructor(webSocketManager: WebSocketManager, subscribeRequest: SubscribeRequest) {
+    super();
+    this.webSocketManager = webSocketManager;
+    this.subscribeRequest = subscribeRequest;
+
+    this.isConnectionLost = false;
+    this.isRestoreFailed = false;
+    this.isSocketReconnected = false;
+    this.isKeepAlive = false;
+
+    this.webSocketManager.addEventListener('message', this.onPing);
```

Comment:
As mentioned, using event listeners here over signals, however this was the only change here.